src/util/log/fd_log.c

```diff
@@ -628,6 +631,82 @@ fd_log_private_2( int          level,
   abort();
 }
 
+char const * 
+fd_log_private_hexdump_msg ( char const * descr,
+                             void const * blob,
+                             ulong        sz ) {
+  if( FD_UNLIKELY( !blob ) ) {
+    FD_LOG_WARNING(( "Hexdump NULL blob ptr" ));
+    return NULL;
+  }
+
+  if( FD_UNLIKELY( !descr ) ) {
+    FD_LOG_WARNING(( "Hexdump NULL blob description" ));
+    return NULL;
+  }
+
+  if( FD_UNLIKELY( strlen( descr )==0 ) ) {
+    FD_LOG_WARNING(( "Hexdump empty blob description" ));
+    return NULL;
+  }
+
+  if( FD_UNLIKELY( strlen( descr )>FD_LOG_HEXDUMP_BLOB_DESCRIPTION_MAX_LEN ) ) {
+    FD_LOG_WARNING(( "Hexdump blob description too long" ));
+    return NULL;
+  }
```

Comment:
See comment around `fd_log.h:172`.  One simple mod would be to do `return strcpy( log_msg, "Hexdump ..." );` in lieu of the log warning.

A more elaborate tweak would be allow for NULL descr and/or truncated descr (pretty printing accordingly) and then logging the case of a NULL blob with a non-zero sz with appropriate error message.

The case of NULL blob with zero sz probably shouldn't be treated as an error case.